### ISSUE TITLE
`dangerous_implicit_autorefs` lint suggestion triggers `clippy::needless_borrow`

### ISSUE THREAD

**Issue by wyfo** [rust-lang/rust#140720]
Let's take the following example:
```rust
unsafe fn slice_ptr_len_because_of_msrv<T>(slice: *[T]) -> usize {
    unsafe { (*slice).len() }
}
```
on current nightly, it triggers:
```
warning: implicit autoref creates a reference to the dereference of a raw pointer
 --> src/main.rs:2:14
  |
3 |     unsafe { (*slice).len() }
  |              ^^^^^^^^^^^^^^
  |
  = note: creating a reference requires the pointer target to be valid and imposes aliasing requirements
  = note: `#[warn(dangerous_implicit_autorefs)]` on by default
help: try using a raw pointer method instead; or if this reference is intentional, make it explicit
  |
3 |     unsafe { (&(*slice)).len() }
  |              ++        +
```
If the suggestion is applied, then clippy complains:
```
warning: this expression borrows a value the compiler would automatically borrow
 --> src/main.rs:2:14
  |
2 |     unsafe { (&(*slice)).len() }
  |              ^^^^^^^^^^^ help: change this to: `(*slice)`
  |
  = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#needless_borrow
  = note: `#[warn(clippy::needless_borrow)]` on by default
```

Curiously enough, `(*slice)[..].len()` also triggers `dangerous_implicit_autorefs`, but the suggested `(&(*slice))[..].len()` doesn't trigger `clippy::needless_borrow`.

**Comment by Urgau** [comment]
This issue is for you @rust-lang/clippy.

*Btw, I'm unable to transfer the issue, would be great to enable triagebot [`[transfer]`](https://forge.rust-lang.org/triagebot/transfer.html) feature in your repo.*

**Comment by wyfo** [comment]
I've opened https://github.com/rust-lang/rust-clippy/issues/14743 instead.

**Comment by fmease** [comment]
Unless the impl has been made more strict since it's introduction, Clippy doesn't need `[transfer]` enabled for `@rustbot transfer rust-clippy` to work from inside r-l/r since only the source repo needs that opt-in (and the target needs to have triagebot enabled which rust-clippy has).

**Comment by Urgau** [comment]
For some reason I though it was the inverse, that rust-lang/rust-clippy needed to have the `[transfer]` feature.
